src/community/hz-cluster/src/main/java/org/geoserver/cluster/hazelcast/EventHzSynchronizer.java

```diff
@@ -198,6 +199,13 @@ private void processCatalogEvent(final ConfigChangeEvent event)
         final Catalog cat = cluster.getRawCatalog();
 
         switch (t) {
+            case PREADD:
+                subj = getCatalogInfo(cat, id, clazz);
+                notifyMethod =
+                        CatalogListener.class.getMethod(
+                                "handlePreAddEvent", CatalogBeforeAddEvent.class);
+                evt = new CatalogAddEventImpl();
+                break;
```

Comment:
The GSIP specify that only the normal GeoServer was targeted, I understand that having events makes it easier to support others, which is good. But a catalog capability shoudl be created for this, we don't know all catalog implementations.

Reply:
understood, I will add default implementation inside interface and revert changes inside community modules